File: Anya_Seth/Sliding Window/Grumpy Bookstore Owner.java
class Solution {
    public int maxSatisfied(int[] customers, int[] grumpy, int minutes) {
        int i = minutes;
        int unsatisfied = 0;
        for(int j = 0 ; j < i ; j++){
            if(grumpy[j] == 1){
               unsatisfied += customers[j];
            }
        }
        int maxValue = unsatisfied;
        while(i < customers.length){
            if(grumpy[i] == 1){
            unsatisfied += customers[i];
            }
            if(grumpy[i-minutes] == 1){
            unsatisfied -= customers[i - minutes];
            }
            maxValue = Math.max(maxValue, unsatisfied);
            i++;
        }
        int ans = maxValue;
        for(int k = 0; k < customers.length ; k++){
            if(grumpy[k] == 0){
            ans += customers[k];
            }
        }
        return ans;
    }
}

File: Lakshay Garg/Leetcode/Sliding Window/Grumpy Bookstore Owner.cpp
class Solution {
public:
    int maxSatisfied(vector<int>& c, vector<int>& g, int m) {
        int n=c.size(),ans=0;
        for(int i=0;i<n;i++){
            if(g[i]==0) ans+=c[i];
        }
        int maxi=0,count=0;
        for(int i=0;i<m;i++){
            if(g[i]==1)count+=c[i];
        }
        maxi=count;
        int start=0,end=m-1;
        while(end<n-1){
            if(g[start]==1) count-=c[start];
            start++;
            end++;
            if(g[end]==1) count+=c[end];
            maxi=max(count,maxi);
        }
        ans+=maxi;
        return  ans;
    }
};
